citations-impl/impl/src/java/org/sakaiproject/citation/impl/DbCitationService.java

```diff
@@ -323,6 +350,304 @@ else if(value instanceof String)
         }
 
 		/* (non-Javadoc)
+		* @see org.sakaiproject.citation.impl.BaseCitationService.Storage#commitCitationCollectionOrder(org.sakaiproject.citation.api.CitationCollectionOrder)
+		*/
+		protected void commitSection(CitationCollectionOrder citationCollectionOrder)
+		{
+			String orderStatement = "insert into " + m_collectionOrderTableName + " (COLLECTION_ID, CITATION_ID, LOCATION, SECTION_TYPE, VALUE) VALUES(?,?,?,?,?)";
+
+			Object[] orderFields = new Object[5];
+			orderFields [0] = citationCollectionOrder.getCollectionId();
+			orderFields [1] = null;
+			orderFields [2] = citationCollectionOrder.getLocation();
+			orderFields [3] = citationCollectionOrder.getSectiontype();
+			orderFields [4] = citationCollectionOrder.getValue();
+
+			m_sqlService.dbWrite(orderStatement, orderFields);
+
+			updateCitationCollectionUpdateDate(citationCollectionOrder.getCollectionId());
+		}
+
+		private void updateCitationCollectionUpdateDate(String citationCollectionId) {
+			String updateStatement = "UPDATE " + m_collectionTableName + " SET PROPERTY_VALUE = ? WHERE COLLECTION_ID= ? AND PROPERTY_NAME = ? ";
+			long mostRecentUpdate = TimeService.newTime().getTime();
+			Object[] updateFields = new Object[3];
+			updateFields[0] = Long.toString(mostRecentUpdate);
+			updateFields[1] = citationCollectionId;
+			updateFields[2] = PROP_MOST_RECENT_UPDATE;
+
+			m_sqlService.dbWrite(updateStatement, updateFields);
+		}
+
+
+		/* (non-Javadoc)
+		* @see org.sakaiproject.citation.impl.BaseCitationService.Storage#commitSubsection(org.sakaiproject.citation.api.CitationCollectionOrder)
+		*/
+		protected void commitSubsection(CitationCollectionOrder citationCollectionOrder)
+		{
+			List<CitationCollectionOrder> citationCollectionOrderList = getNestedCollectionAsList(citationCollectionOrder.getCollectionId());
+
+			// insert subsection into the right position in the tree
+			int subSectionLocation = 1;
+			List<CitationCollectionOrder> citationCollectionOrderToSave = new ArrayList<CitationCollectionOrder>();
+			for (CitationCollectionOrder collectionOrder : citationCollectionOrderList) {
+				if (citationCollectionOrder.getLocation() == subSectionLocation){
+					citationCollectionOrderToSave.add(citationCollectionOrder);
+				}
+				citationCollectionOrderToSave.add(collectionOrder);
+				subSectionLocation++;
+			}
+			if (citationCollectionOrderToSave.size()==citationCollectionOrderList.size()){
+				citationCollectionOrderToSave.add(citationCollectionOrder);
+			}
+
+			// delete and insert nested sections atomically
+			Connection conn = null;
```

Comment:
commitCitationCollectionOrders(List<CitationCollectionOrder> citationCollectionOrders, String citationCollectionId) also needs to be wrapped in a transaction